src/Component.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\SnowflakeParametersExtractor;
+
+use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvFile;
+use Keboola\Db\Import\Exception as SnowflakeException;
+use Keboola\Db\Import\Snowflake\Connection;
+
+class Component extends BaseComponent
+{
+    public function run(): void
+    {
+        $parametersFetcher = new ParametersFetcher($this->createSnowflakeConnection());
+        $parameters = $this->addHostToParameters(
+            $parametersFetcher->fetchAccountParameters()
+        );
+
+        $csvFile = new CsvFile($this->getAccountParametersTableDestination());
+        foreach ($parameters as $parameter) {
+            $csvFile->writeRow($parameter);
+        }
+
+        $this->getManifestManager()->writeTableManifestFromArray(
+            $this->getAccountParametersTableDestination(),
+            [
+                'primary_key' => [
+                    'host',
+                    'key',
+                ],
+                'columns' => array_keys($parameters[0]),
+            ]
+        );
+    }
+
+    private function addHostToParameters(array $parameters): array
```

Comment:
co kdyby tohle řešil procesor?

Reply:
Teoreticky by mohl být "obecný" třeba "join files", který by umožnil přidat název souboru jako sloupec. Ale mít na to dedikovaný processor mi přijde jako overengineering.

Reply:
me to prijde taky jako overengineering, tohle je ficura komponenty, procesor je nepovinnej (cimz nerikam, ze addcolumn procesor by nemohl existovat)

Reply:
netušíte, proč ta query nejde stáhnout obyč Snowflake extraktorem?

https://connection.keboola.com/admin/projects/232/jobs/375285902

Reply:
Jestli to je kvůli oprávnění (je tam normální user pro writer), tak bych čekal jinou hlášku :-).

Reply:
ne není to opravněním, viz. link na diskuzi nahoře

Reply:
je to tím že snflk extractor je dělaný na tahání tabulek a obaluje to ještě selectem pro detekci sloupců a metadat.

Reply:
a rozhodně jsem to nechtěl bastlit do něj, to radši udělám jednoúčelovou věc jako tohle, vše je v té diskuzi

Reply:
nebo možná nerozumím tomu co teď zkoumáš? snlk extractor byla první věc kteoru jsem zkoušel